pkg/controller/build/buildrequest/buildrequest.go

```diff
@@ -198,6 +218,92 @@ func (br buildRequestImpl) additionaltrustbundleToConfigMap() *corev1.ConfigMap
 	return configmap
 }
 
+func (br buildRequestImpl) etcPolicyToConfigMap(mc *mcfgv1.MachineConfig) (*corev1.ConfigMap, error) {
+	// Build the ConfigMap data
+	configMapData, err := br.getConfigMapData(mc, "/etc/containers/policy.json", "/etc/containers/")
+	if err != nil {
+		return nil, err
+	}
+	if len(configMapData) == 0 {
+		return nil, nil
+	}
+
+	// Create the ConfigMap
+	configmap := &corev1.ConfigMap{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: br.getObjectMeta(br.getEtcPolicyConfigMapName()),
+		Data:       configMapData,
+	}
+	return configmap, nil
+}
+
+func (br buildRequestImpl) etcRegistriesToConfigMap(mc *mcfgv1.MachineConfig) (*corev1.ConfigMap, error) {
+	// Build the ConfigMap data
+	configMapData, err := br.getConfigMapData(mc, "/etc/containers/registries.conf", "/etc/containers/")
+	if err != nil {
+		return nil, err
+	}
+	if len(configMapData) == 0 {
+		return nil, nil
+	}
+
+	// Create the ConfigMap
+	configmap := &corev1.ConfigMap{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: br.getObjectMeta(br.getEtcRegistriesConfigMapName()),
+		Data:       configMapData,
+	}
+	return configmap, nil
+}
+
+func (br buildRequestImpl) getConfigMapData(mc *mcfgv1.MachineConfig, filePath, prefixToTrim string) (map[string]string, error) {
+	if len(mc.Spec.Config.Raw) == 0 {
+		return nil, nil
+	}
+	// Build the ConfigMap data
+	ignCfg, err := ctrlcommon.ParseAndConvertConfig(mc.Spec.Config.Raw)
+	if err != nil {
+		return nil, fmt.Errorf("parsing rendered MC Ignition config failed with error: %w", err)
+	}
+
+	for _, file := range ignCfg.Storage.Files {
+		if file.Path != filePath {
+			continue
+		}
+		if file.Contents.Source == nil {
+			return nil, fmt.Errorf("nil source for %s", file.Path)
+		}
+
+		// Extract and decode the encoded data
```

Comment:
thought (non-blocking): A simpler way to decode this content is to use this helper function: https://github.com/openshift/machine-config-operator/blob/main/pkg/controller/common/helpers.go#L620

Reply:
Fixed, thanks!